### PR TITLE
fix: use updated Git API

### PR DESCRIPTION
**Pull request by mattrothenberg**
As reported in https://github.com/githubocto/flat-editor/issues/23, folks are having trouble opening Flat Editor and it's crashing with a Git error.

After digging into (and reproducing locally) the issue, it seems like the Git VSCode extension API changed somewhere along the way, and some of the methods we were using no longer returned the correct data.

I've added the [latest and greatest Git type definitions from VSCode](https://github.com/microsoft/vscode/blob/main/extensions/git/src/api/git.d.ts) and removed a few unused paths in our VSCodeGit class

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -34,6 +34,9 @@
       "@babel/plugin-transform-runtime"
     ]
   },
+  "extensionDependencies": [
+    "vscode.git"
+  ],
```
Comment (mattrothenberg) [inline package.json:39]: While this isn't strictly necessary, I like calling out explicitly that we're using the native Git extension.

File: src/flatConfigEditor.ts [hunk 2]
```diff
@@ -422,12 +422,8 @@ export class FlatConfigEditor implements vscode.CustomTextEditorProvider {
     return new Promise(async (resolve, reject) => {
       try {
         const gitClient = new VSCodeGit()
-        await gitClient.activateExtension()
-        await gitClient.waitForRepo(3)
-
-        // Next, let's grab the repo name.
+        await gitClient.waitForRepo()
```
Comment (mattrothenberg) [inline src/flatConfigEditor.ts:425]: We only ever call `waitForRepo` here... As such, I removed the "# of times" argument from the function and co-located it in the function implementation itself.

File: src/git.ts [hunk 3]
```diff
@@ -1,47 +1,36 @@
...
-  }
-
-  waitForRepo(times: number): Promise<{ name: string; owner: string }> {
+  waitForRepo(): Promise<{ name: string; owner: string }> {
     let count = 0
 
     return new Promise((resolve, reject) => {
       let interval = setInterval(() => {
         try {
-          const remotes = this.repository._repository.remotes
```
Comment (mattrothenberg) [inline src/git.ts:38]: This was no longer returning the right data

File: src/types.ts [hunk 4]
```diff
@@ -118,73 +118,3 @@ export type FlatYamlDoc = {
     scheduled: FlatYamlJob
   }
 }
-
-export interface CommitOptions {
-  all?: boolean | 'tracked'
-}
-
```
Comment (mattrothenberg) [inline src/types.ts:125]: Outdated type defs.

**Review by jaked — APPROVED** [review]
approving on faith—cool to see a little of how it works!

**Pull request merged**